section_1_review.livemd

```diff
@@ -290,7 +333,10 @@ defmodule Recursion do
   iex> Recursion.sum([1, 2, 3])
   6
   """
-  def sum(list) do
+  def sum([]), do: 0
```

Comment:
Very elegant.

Note that this will be body recursion, rather than tail recursion. Likely that's not an issue, but something to be aware of.

To use tail recursion, you could store an accumulator using a second parameter.